File: MORM.Java/src/MORM/Java/Classes/Conexao.java
package MORM.Java.Classes;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;

public class Conexao
{
    public Conexao(Parametro parametro)
    {
        Parametro = parametro;
        
        ConnectionString = 
        	ProviderFactory.GetConnectionStringByProvider(parametro.ProviderName)
        	.replace("@database", parametro.Database)
        	.replace("@username", parametro.Username)
        	.replace("@password", parametro.Password)
        	.replace("@hostname", parametro.Hostname);
        
        Connection = ProviderFactory.CreateConnection(parametro.ProviderName, ConnectionString, parametro.Username, parametro.Password);
        //Connection.open();
    }
    
    public Parametro Parametro;
    public String ConnectionString; 
    public Connection Connection;
    
    public ResultSet GetConsulta(String sql)
    {
    	ResultSet resultSet = null;
		try {
			Statement stm = Connection.createStatement();
			resultSet = stm.executeQuery(sql);
		} catch (SQLException e) {
			e.printStackTrace();
		}
		return resultSet;
    }
    
    public void ExecComando(String cmd)
    {
		try {
			Statement stm = Connection.createStatement();
			stm.execute(cmd);
		} catch (SQLException e) {
			e.printStackTrace();
		}
    }
}

File: MORM.CSharp/MORM.Data/Classes/Conexao.cs
using System.Data.Common;

namespace MORM.Data.Classes
{
    public class Conexao
    {
        public Conexao(Parametro parametro)
        {
            Parametro = parametro;
            
            ConnectionString = 
                ProviderFactory.GetConnectionStringByProvider(parametro.ProviderName)
                .Replace("@database", parametro.Database)
                .Replace("@username", parametro.Username)
                .Replace("@password", parametro.Password)
                .Replace("@hostname", parametro.Hostname);
            
            Connection = ProviderFactory.CreateDbConnection(parametro.ProviderName, ConnectionString);
            Connection.Open();    
        }
        
        public Parametro Parametro { get; private set; }
        public string ConnectionString { get; private set; }
        public DbConnection Connection { get; private set; }
        
        public void ExecComando(string cmd)
        {
            var command = Connection.CreateCommand();
            command.CommandText = cmd;
            command.ExecuteNonQuery();
        }

        public DbDataReader GetConsulta(string sql)
        {
            var command = Connection.CreateCommand();
            command.CommandText = sql;
            return command.ExecuteReader();
        }        
    }
}